codebuild/bin/install_al2_dependencies.sh

```diff
@@ -16,12 +16,47 @@
 set -eu
 source ./codebuild/bin/s2n_setup_env.sh
 
-if [[ ${DISTRO} != "amazon linux" ]]; then
-    echo "Target AL2, but running on $DISTRO: Nothing to do."
-    exit 0
-fi
+al2023_main(){
+    echo "Insatlling AL2023 packages"
+    common_packages
+    al2023_packages
+    versions
+}
+
+al2_main() {
+    echo "Insatlling AL2 packages"
+    common_packages
+    al2_packages
+    symlink_all_the_things
+
+    case "$S2N_LIBCRYPTO" in
+    "openssl-1.1.1")
+        yum erase -y openssl-devel || true
+        yum install -y openssl11-static openssl11-libs openssl11-devel
+        ;;
+    "default") echo "Using default system libcrypto";;
+    *) echo "Unknown libcrypto: ${S2N_LIBCRYPTO}"; exit 1;;
```

Comment:
Should the AL2023 path do a similar check to make sure that openssl 3 was intended? Or is there a different check that will fail if a different libcrypto is specified?

Reply:
Good call, no other checks exist; cmake will just fail, so I can add a more friendly error.